src/io/utils.js

```diff
@@ -14,32 +14,27 @@ export function walkPromise(curPath, { shouldIncludePath = () => true } = {}) {
   // so all file paths (the result keys) can
   // be relative to the starting point.
   const basePath = curPath;
+  // eslint-disable-next-line consistent-return
```

Comment:
If I'm reading this walk function correctly, we should be able to remove this `eslint-disable-next-line` comment by remove the `return result` from line 24 and 37 and add a single `return result` outsitde of the `if (...) {...} else if (...) ...`.

(I'm also wondering if we should raise an error if none of the conditions are met)

Reply:
@rpl I made one return statement after if-else block, but I don't know about raising an error...As I understand we check if it's file and if it's directory... but what else could it be?

Reply:
When the files are coming directly from the filesystem (instead of coming from the content of a zip file, e.g. for an xpi or crx file), the stat result could also be a block or character device, or a FIFO or a socket (based on the related nodejs docs: https://nodejs.org/api/fs.html#fs_class_fs_stats), which is unexpected but still possible.

Currently we just silently ignore these files, and so I'm wondering if it would be better to produce a warning or an error to make it more visible to the addon developer.

Anyway, I'm also ok to defer this to a follow up (so that we keep the changes related to the async/await refactoring separated from changes of the addons-linter behaviors).

@EnTeQuAk what do you think about this?